diff --git a/gme/Fir_Resampler.cpp b/gme/Fir_Resampler.cpp
index d8dd683..ae5f2df 100644
--- a/gme/Fir_Resampler.cpp
+++ b/gme/Fir_Resampler.cpp
@@ -188,6 +188,8 @@ int Fir_Resampler_::skip_input( long count )
 {
 	int remain = write_pos - buf.begin();
 	int max_count = remain - width_ * stereo;
+	if ( max_count < 0 )
+		return 0;
 	if ( count > max_count )
 		count = max_count;
 	
diff --git a/gme/blargg_source.h b/gme/blargg_source.h
index deea918..719f6ab 100644
--- a/gme/blargg_source.h
+++ b/gme/blargg_source.h
@@ -34,7 +34,7 @@
 /* Like printf() except output goes to debug log file. Might be defined to do
  * nothing (not even evaluate its arguments).
  * void debug_printf( const char* format, ... ); */
-#if defined(__cplusplus) && defined(BLARGG_BUILD_DLL)
+#if defined(__cplusplus)
     static inline void blargg_dprintf_( const char* fmt_str, ... ) { (void) fmt_str; }
     #undef debug_printf
     #define debug_printf (1) ? (void) 0 : blargg_dprintf_
